prometheus/registry.go

```diff
@@ -426,7 +426,7 @@ func (r *registry) writePB(w io.Writer, writeEncoded encoder) (int, error) {
 
 	// Drain metricChan in case of premature return.
 	defer func() {
-		for _ = range metricChan {
+		for range metricChan {
```

Comment:
We said a while ago that we don't want to require users of the lib to be on Go >=1.4, so this kind of syntactic change around the `range` statement (automatically done by gofmt) would need to be reverted for older Go versions to work with this library.